device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/fileupload/FileUploadTaskTest.java

```diff
@@ -764,4 +771,67 @@ public void runSendNotificationThrows() throws IOException, IllegalArgumentExcep
         };
     }
 
+    @Test
+    public void getFileUploadSasUriChecksHttpStatusCode() throws IOException, IllegalArgumentException
+    {
+        // arrange
+        FileUploadTask fileUploadTask = Deencapsulation.newInstance(FileUploadTask.class,
+            new Class[] {String.class, InputStream.class, long.class, HttpsTransportManager.class, IotHubEventCallback.class, Object.class},
+            VALID_BLOB_NAME, mockInputStream, VALID_STREAM_LENGTH, mockHttpsTransportManager, mockIotHubEventCallback, VALID_CALLBACK_CONTEXT);
+
+        new Expectations()
+        {
+            {
+                mockHttpsTransportManager.getFileUploadSasUri((IotHubTransportMessage) any);
+                result = mockResponseMessage;
+
+                mockResponseMessage.getStatus();
+                result = IotHubStatusCode.UNAUTHORIZED;
```

Comment:
this is just arbitrary for the test, right?

Reply:
Yeah